Silex 2 compatibility

roest01 opened the pull request:
Hey Timo,

Thanks for this package, we used it and migrate the code to fit with silex 2, Guzzle 6 and Guzzle oauth subscriber 0.3. I offer it back to you with this PR.

As you see our branch contains major changes. We have double test it, but if you see any trouble within the code don't hesitate to remark it.

best Tobias

roest01 committed:
migrate lib to work with silex 2.0

- Used new ControllerProviderInterface
- replaced share function with normal function because share got removed
- implement BootableProviderInterface in JiraOAuthServiceProvider
- use Pimple ServiceProviderInterface instead of Silex 1.3 Interface

roest01 committed:
modify ServiceProvider to fit with Guzzle 6

- use new lira callback url schema with absolute url
- add configurable private_key_passphrase to config
- rename base_url to base_uri because renamed in Guzzle 6
- replaced removed Guzzle emitter with Guzzle HandlerStack
- remove file:// prefix for private key because of waste

elmi82 commented:
@roest01: Thanks for the patch, I think it looks good. Unfortunately I don't have access to Jira anymore. @michaelknapp Do you have time to test this?

roest01 committed:
update documentation; fix some stuff

- remove guzzle oauth minimum stability from readme because not
necessary anymore
- add private_key_passphrase to read
- update usage documentation in readme
- rename base_uri back to base_url because previously was better
- implement more secure way to detect https callback
